### DISCUSSION TITLE
Conflicting flake8 dependency in Airflow 2.2.2

### DISCUSSION THREAD

**Discussion by mitsos1os** [apache/airflow#25213] [General]
### Apache Airflow version

2.2.2

### What happened

Hi, we are using the managed MWAA 2.2.2 version of Airflow. We have set up our requirements installation to use the constraints file from [here](https://raw.githubusercontent.com/apache/airflow/constraints-2.2.2/constraints-3.7.txt)

When we try adding **flake8** to our local requirements setup (using the same constraint file), we are getting a conflicting dependency issue and package is not installed.

More specifically:

> ERROR: Cannot install -r requirements-local.txt (line 14) because these package versions have conflicting dependencies.
> 
> The conflict is caused by:
>     flake8 3.9.2 depends on pycodestyle<2.8.0 and >=2.7.0
>     The user requested (constraint) pycodestyle==2.8.0

This is happening, because the linked constraints file is explicitly using `flake8==3.9.2` and at the same time `pycodestyle==2.8.0` versions. However **flake8** 3.9.2 is not compatible with 2.8.0 version of **pycodestyle** according to the error above.

### What you think should happen instead

The dependencies referenced in the constraints should be installable without any conflicts

### How to reproduce

example **requirements.txt** file

```
--constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.2.2/constraints-3.7.txt

apache-airflow[cncf.kubernetes]==2.2.2
flake8
```

Then just run: `pip install -r requirements.txt`

### Operating System

Ubuntu 20.04.4 LTS

### Versions of Apache Airflow Providers

```
apache-airflow-providers-amazon==2.4.0
apache-airflow-providers-cncf-kubernetes==2.1.0
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-http==2.0.1
apache-airflow-providers-imap==2.0.1
apache-airflow-providers-sqlite==2.0.1
```

### Deployment

MWAA

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
You likely made a mistake by using wrong python version

**Reply by potiuk** [reply]
constraints-3.7.txt -> Python 3.7

**Reply by mitsos1os** [reply]
No, I am using python 3.7.13
Also you can see what I am talking about directly in the constraints file (about the actual conflicting versions)

You can even try to install it independently with ` pip install -c https://raw.githubusercontent.com/apache/airflow/constraints-2.2.2/constraints-3.7.txt flake8==3.9.2` and get the same error

**Reply by potiuk** [reply]
Actually not  - the problem is that thos are "devel" dependencies which do not always follow constraints. Simply remove constraints when installing devel dependencie like that. They are not used by running airflow, so should have no impact on the running installation

**Reply by potiuk** [reply]
Just remove constraints when installing them -> issue solved.

**Reply by mitsos1os** [reply]
ok then. Thanks!
